# Data Analysis Project

The Data of different cars is given with their specifications.

In [2]:
import pandas as pd

The data is available as a CSV file.Analyzing this data set using Pandas DataFrame.

In [65]:
car=pd.read_csv(r'C:\Users\ayan5\Desktop\Cars Data For project\2. Cars Data1.csv')

In [66]:
car.head()

Make           Model   Type Origin DriveTrain      MSRP   Invoice  \
0  Acura             MDX    SUV   Asia        All  $36,945   $33,337    
1  Acura  RSX Type S 2dr  Sedan   Asia      Front  $23,820   $21,761    
2  Acura         TSX 4dr  Sedan   Asia      Front  $26,990   $24,647    
3  Acura          TL 4dr  Sedan   Asia      Front  $33,195   $30,299    
4  Acura      3.5 RL 4dr  Sedan   Asia      Front  $43,755   $39,014    

   EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  Weight  \
0         3.5        6.0         265        17           23    4451   
1         2.0        4.0         200        24           31    2778   
2         2.4        4.0         200        22           29    3230   
3         3.2        6.0         270        20           28    3575   
4         3.5        6.0         225        18           24    3880   

   Wheelbase  Length  
0        106     189  
1        101     172  
2        105     183  
3        108     186  
4        115     197

In [67]:
car.shape

(428, 15)

So there are total 428 rows and 15 columns in dataset.

# Data Cleaning

Find all the null value in the dataset.If there is any null value in any column,then fill it with the mean of that column and for non-numerical column fill it with mode of that column.

In [68]:
car.isnull().sum()

Make           0
Model          0
Type           0
Origin         0
DriveTrain     0
MSRP           0
Invoice        0
EngineSize     0
Cylinders      2
Horsepower     0
MPG_City       0
MPG_Highway    0
Weight         0
Wheelbase      0
Length         0
dtype: int64

In 'cylinder' column only 2 null values are there.

In [72]:
#To replace null values in cylinder column
car['Cylinders'].fillna(car['Cylinders'].mean(), inplace=True)

In [73]:
car.isnull().sum()

Make           0
Model          0
Type           0
Origin         0
DriveTrain     0
MSRP           0
Invoice        0
EngineSize     0
Cylinders      0
Horsepower     0
MPG_City       0
MPG_Highway    0
Weight         0
Wheelbase      0
Length         0
dtype: int64

Now here all null values replaced.

# Value Counts

Check what are the different types of make are there in our dataset and what is the count(occurence) of each Make in the data in the data?

In [74]:
car.head(5)

Make           Model   Type Origin DriveTrain      MSRP   Invoice  \
0  Acura             MDX    SUV   Asia        All  $36,945   $33,337    
1  Acura  RSX Type S 2dr  Sedan   Asia      Front  $23,820   $21,761    
2  Acura         TSX 4dr  Sedan   Asia      Front  $26,990   $24,647    
3  Acura          TL 4dr  Sedan   Asia      Front  $33,195   $30,299    
4  Acura      3.5 RL 4dr  Sedan   Asia      Front  $43,755   $39,014    

   EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  Weight  \
0         3.5        6.0         265        17           23    4451   
1         2.0        4.0         200        24           31    2778   
2         2.4        4.0         200        22           29    3230   
3         3.2        6.0         270        20           28    3575   
4         3.5        6.0         225        18           24    3880   

   Wheelbase  Length  
0        106     189  
1        101     172  
2        105     183  
3        108     186  
4        115     197

In [75]:
car['Make'].value_counts()

Toyota           28
Chevrolet        27
Mercedes-Benz    26
Ford             23
BMW              20
Audi             19
Honda            17
Nissan           17
Volkswagen       15
Chrysler         15
Dodge            13
Mitsubishi       13
Volvo            12
Jaguar           12
Hyundai          12
Subaru           11
Pontiac          11
Mazda            11
Lexus            11
Kia              11
Buick             9
Mercury           9
Lincoln           9
Saturn            8
Cadillac          8
Suzuki            8
Infiniti          8
GMC               8
Acura             7
Porsche           7
Saab              7
Land Rover        3
Oldsmobile        3
Jeep              3
Scion             2
Isuzu             2
MINI              2
Hummer            1
Name: Make, dtype: int64







# Filtering

Show all the records where origin is Asia Or Europe.

In [76]:
car.head(6)

Make                    Model   Type Origin DriveTrain      MSRP  \
0  Acura                      MDX    SUV   Asia        All  $36,945    
1  Acura           RSX Type S 2dr  Sedan   Asia      Front  $23,820    
2  Acura                  TSX 4dr  Sedan   Asia      Front  $26,990    
3  Acura                   TL 4dr  Sedan   Asia      Front  $33,195    
4  Acura               3.5 RL 4dr  Sedan   Asia      Front  $43,755    
5  Acura  3.5 RL w/Navigation 4dr  Sedan   Asia      Front  $46,100    

    Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  Weight  \
0  $33,337          3.5        6.0         265        17           23    4451   
1  $21,761          2.0        4.0         200        24           31    2778   
2  $24,647          2.4        4.0         200        22           29    3230   
3  $30,299          3.2        6.0         270        20           28    3575   
4  $39,014          3.5        6.0         225        18           24    3880   
5  $41,100          3.5        6.0         225        18           24    3893   

   Wheelbase  Length  
0        106     189  
1        101     172  
2        105     183  
3        108     186  
4        115     197  
5        115     197

In [77]:
car[car['Origin'].isin(['Asia','Europe'])]

Make                    Model   Type  Origin DriveTrain      MSRP  \
0    Acura                      MDX    SUV    Asia        All  $36,945    
1    Acura           RSX Type S 2dr  Sedan    Asia      Front  $23,820    
2    Acura                  TSX 4dr  Sedan    Asia      Front  $26,990    
3    Acura                   TL 4dr  Sedan    Asia      Front  $33,195    
4    Acura               3.5 RL 4dr  Sedan    Asia      Front  $43,755    
..     ...                      ...    ...     ...        ...       ...   
423  Volvo  C70 LPT convertible 2dr  Sedan  Europe      Front  $40,565    
424  Volvo  C70 HPT convertible 2dr  Sedan  Europe      Front  $42,565    
425  Volvo               S80 T6 4dr  Sedan  Europe      Front  $45,210    
426  Volvo                      V40  Wagon  Europe      Front  $26,135    
427  Volvo                     XC70  Wagon  Europe        All  $35,145    

      Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  \
0    $33,337          3.5        6.0         265        17           23   
1    $21,761          2.0        4.0         200        24           31   
2    $24,647          2.4        4.0         200        22           29   
3    $30,299          3.2        6.0         270        20           28   
4    $39,014          3.5        6.0         225        18           24   
..        ...         ...        ...         ...       ...          ...   
423  $38,203          2.4        5.0         197        21           28   
424  $40,083          2.3        5.0         242        20           26   
425  $42,573          2.9        6.0         268        19           26   
426  $24,641          1.9        4.0         170        22           29   
427  $33,112          2.5        5.0         208        20           27   

     Weight  Wheelbase  Length  
0      4451        106     189  
1      2778        101     172  
2      3230        105     183  
3      3575        108     186  
4      3880        115     197  
..      ...        ...     ...  
423    3450        105     186  
424    3450        105     186  
425    3653        110     190  
426    2822        101     180  
427    3823        109     186  

[281 rows x 15 columns]

# Removing Unwanted Records

Remove all the records (rows) where weight is above 4000.

In [78]:
car.head(6)

Make                    Model   Type Origin DriveTrain      MSRP  \
0  Acura                      MDX    SUV   Asia        All  $36,945    
1  Acura           RSX Type S 2dr  Sedan   Asia      Front  $23,820    
2  Acura                  TSX 4dr  Sedan   Asia      Front  $26,990    
3  Acura                   TL 4dr  Sedan   Asia      Front  $33,195    
4  Acura               3.5 RL 4dr  Sedan   Asia      Front  $43,755    
5  Acura  3.5 RL w/Navigation 4dr  Sedan   Asia      Front  $46,100    

    Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  Weight  \
0  $33,337          3.5        6.0         265        17           23    4451   
1  $21,761          2.0        4.0         200        24           31    2778   
2  $24,647          2.4        4.0         200        22           29    3230   
3  $30,299          3.2        6.0         270        20           28    3575   
4  $39,014          3.5        6.0         225        18           24    3880   
5  $41,100          3.5        6.0         225        18           24    3893   

   Wheelbase  Length  
0        106     189  
1        101     172  
2        105     183  
3        108     186  
4        115     197  
5        115     197

In [79]:
car[~(car['Weight']>4000)]

Make                    Model   Type  Origin DriveTrain      MSRP  \
1    Acura           RSX Type S 2dr  Sedan    Asia      Front  $23,820    
2    Acura                  TSX 4dr  Sedan    Asia      Front  $26,990    
3    Acura                   TL 4dr  Sedan    Asia      Front  $33,195    
4    Acura               3.5 RL 4dr  Sedan    Asia      Front  $43,755    
5    Acura  3.5 RL w/Navigation 4dr  Sedan    Asia      Front  $46,100    
..     ...                      ...    ...     ...        ...       ...   
423  Volvo  C70 LPT convertible 2dr  Sedan  Europe      Front  $40,565    
424  Volvo  C70 HPT convertible 2dr  Sedan  Europe      Front  $42,565    
425  Volvo               S80 T6 4dr  Sedan  Europe      Front  $45,210    
426  Volvo                      V40  Wagon  Europe      Front  $26,135    
427  Volvo                     XC70  Wagon  Europe        All  $35,145    

      Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  \
1    $21,761          2.0        4.0         200        24           31   
2    $24,647          2.4        4.0         200        22           29   
3    $30,299          3.2        6.0         270        20           28   
4    $39,014          3.5        6.0         225        18           24   
5    $41,100          3.5        6.0         225        18           24   
..        ...         ...        ...         ...       ...          ...   
423  $38,203          2.4        5.0         197        21           28   
424  $40,083          2.3        5.0         242        20           26   
425  $42,573          2.9        6.0         268        19           26   
426  $24,641          1.9        4.0         170        22           29   
427  $33,112          2.5        5.0         208        20           27   

     Weight  Wheelbase  Length  
1      2778        101     172  
2      3230        105     183  
3      3575        108     186  
4      3880        115     197  
5      3893        115     197  
..      ...        ...     ...  
423    3450        105     186  
424    3450        105     186  
425    3653        110     190  
426    2822        101     180  
427    3823        109     186  

[325 rows x 15 columns]

# Applying Function on a column

Increase all the values of 'MPG_City' column by 3.

In [81]:
car.head(6)

Make                    Model   Type Origin DriveTrain      MSRP  \
0  Acura                      MDX    SUV   Asia        All  $36,945    
1  Acura           RSX Type S 2dr  Sedan   Asia      Front  $23,820    
2  Acura                  TSX 4dr  Sedan   Asia      Front  $26,990    
3  Acura                   TL 4dr  Sedan   Asia      Front  $33,195    
4  Acura               3.5 RL 4dr  Sedan   Asia      Front  $43,755    
5  Acura  3.5 RL w/Navigation 4dr  Sedan   Asia      Front  $46,100    

    Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  Weight  \
0  $33,337          3.5        6.0         265        17           23    4451   
1  $21,761          2.0        4.0         200        24           31    2778   
2  $24,647          2.4        4.0         200        22           29    3230   
3  $30,299          3.2        6.0         270        20           28    3575   
4  $39,014          3.5        6.0         225        18           24    3880   
5  $41,100          3.5        6.0         225        18           24    3893   

   Wheelbase  Length  
0        106     189  
1        101     172  
2        105     183  
3        108     186  
4        115     197  
5        115     197

In [82]:
car['MPG_City']=car['MPG_City'].apply(lambda a:a+3)

In [38]:
car

Make                    Model   Type  Origin DriveTrain      MSRP  \
0    Acura                      MDX    SUV    Asia        All  $36,945    
1    Acura           RSX Type S 2dr  Sedan    Asia      Front  $23,820    
2    Acura                  TSX 4dr  Sedan    Asia      Front  $26,990    
3    Acura                   TL 4dr  Sedan    Asia      Front  $33,195    
4    Acura               3.5 RL 4dr  Sedan    Asia      Front  $43,755    
..     ...                      ...    ...     ...        ...       ...   
427  Volvo  C70 LPT convertible 2dr  Sedan  Europe      Front  $40,565    
428  Volvo  C70 HPT convertible 2dr  Sedan  Europe      Front  $42,565    
429  Volvo               S80 T6 4dr  Sedan  Europe      Front  $45,210    
430  Volvo                      V40  Wagon  Europe      Front  $26,135    
431  Volvo                     XC70  Wagon  Europe        All  $35,145    

      Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  \
0    $33,337          3.5        6.0       265.0      20.0         23.0   
1    $21,761          2.0        4.0       200.0      27.0         31.0   
2    $24,647          2.4        4.0       200.0      25.0         29.0   
3    $30,299          3.2        6.0       270.0      23.0         28.0   
4    $39,014          3.5        6.0       225.0      21.0         24.0   
..        ...         ...        ...         ...       ...          ...   
427  $38,203          2.4        5.0       197.0      24.0         28.0   
428  $40,083          2.3        5.0       242.0      23.0         26.0   
429  $42,573          2.9        6.0       268.0      22.0         26.0   
430  $24,641          1.9        4.0       170.0      25.0         29.0   
431  $33,112          2.5        5.0       208.0      23.0         27.0   

     Weight  Wheelbase  Length origin  
0    4451.0      106.0   189.0   Asia  
1    2778.0      101.0   172.0   Asia  
2    3230.0      105.0   183.0   Asia  
3    3575.0      108.0   186.0   Asia  
4    3880.0      115.0   197.0   Asia  
..      ...        ...     ...    ...  
427  3450.0      105.0   186.0   Asia  
428  3450.0      105.0   186.0   Asia  
429  3653.0      110.0   190.0   Asia  
430  2822.0      101.0   180.0   Asia  
431  3823.0      109.0   186.0   Asia  

[432 rows x 16 columns]